src/indieweb_utils.py

```diff
@@ -345,28 +345,37 @@ def discover_web_page_feeds(url):
 
     # check for presence of mf2 hfeed
     h_feed = soup.find_all(class_="h-feed")
+    page_title = soup.find("title")
+
+    page_domain = url.split("/")[2]
+
+    valid_mime_types = [
+        "application/rss+xml",
+        "application/atom+xml",
+        "application/rdf+xml",
+        "application/xml",
+        "application/json",
+        "application/mf2+json",
+        "application/atom+xml",
+        "application/feed+json",
+        "application/jf2feed_json",
+    ]
 
-    feeds = []
+    valid_mime_types += user_mime_types
 
-    if soup.find("link", rel="alternate", type="application/atom+xml"):
-        feeds.append(soup.find("link", rel="alternate", type="application/atom+xml").get("href"))
-    if soup.find("link", rel="alternate", type="application/rss+xml"):
-        feeds.append(soup.find("link", rel="alternate", type="application/rss+xml").get("href"))
-    if soup.find("link", rel="feed", type="text/html"):
-        # used for mircoformats rel=feed discovery
-        feeds.append(soup.find("link", rel="feed", type="text/html").get("href"))
-    if h_feed:
-        feeds.append(url)
+    valid_mime_types = list(set(valid_mime_types))
 
-    for feed in range(len(feeds)):
-        f = feeds[feed]
+    feeds = {}
 
-        if f.startswith("/"):
-            feeds[feed] = url.strip("/") + f
-        elif f.startswith("http://") or f.startswith("https://"):
-            pass
-        elif f.startswith("//"):
-            feeds[feed] = "https:" + f
+    for mime_type in valid_mime_types:
+        if soup.find("link", rel="alternate", type=mime_type):
+            feed_title = soup.find("link", rel="alternate", type=mime_type).get("title")
```

Comment:
Can feed_title be `None`? We may want to normalize as an empty string if no value is found so we can simplify working with the returned data i.e. the title is always a string.